presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -465,14 +464,21 @@ public synchronized void renameTable(String databaseName, String tableName, Stri
         requireNonNull(newDatabaseName, "newDatabaseName is null");
         requireNonNull(newTableName, "newTableName is null");
 
-        getRequiredTable(databaseName, tableName);
+        Table table = getRequiredTable(databaseName, tableName);
         getRequiredDatabase(newDatabaseName);
 
+        if (isIcebergTable(table.getParameters())) {
```

Comment:
We should be able to support rename for iceberg. Iceberg it self does not really have any table name property, it just has a table location so we should be able to rename the table in hive metastore while keeping the same table location.

Reply:
The file metastore is based on the directory structure. For example, listing table names will list the schema directory. So if we rename the table (and thus the directory), all the Iceberg manifest files will still point at the old location.

We could change the design of the file metastore to separate the metadata and data files, but it’s mostly used for testing now.